experimental/JuLie/schur_polynomials.jl

```diff
@@ -10,24 +10,18 @@
 export schur_polynomial
 
 @doc Markdown.doc"""
-    schur_polynomial(lambda::Partition{T}, n::Int=sum(lambda); ring::FmpzMPolyRing) where T<:Integer
-    schur_polynomial(R::FmpzMPolyRing, lambda::Partition{T}, n::Int=sum(lambda)) where T<:Integer
-    schur_polynomial(lambda::Partition{T}, x::Vector{fmpz_mpoly}) where T<:Integer
```

Comment:
@ulthiel I removed `schur_polynomial(lambda, x)` because it was buggy (the content of `x` was ignored in many cases, it only worked right when `x == gens(R)`.

If supporting this for an arbitrary vector of polynomials is important to you, we can of course also keep it but then should fix it. I can look into doing that but I did not want to expend the effort if nobody then intends to use it ;-).

Reply:
I have to remember what was being done here. I believe that passing the vector should return the evaluated Schur polynomial in this vector. This should be faster than first computing the Schur polynomial (in the indeterminates) and then evaluating. So I believe there's good reason for this function in principle.

Reply:
Yes, `schur_polynomial(lambda, x)` as it is in the code is useless and can be removed. In principle there would be a purpose (when x is a vector of elements of a ring) but this needs to be implemented separately I think. Now there's still `schur_polynomial_cbf(lambda, x)` where the x is technically not necessary as well and should better have the same signature as the other functions (needs some rewriting of the code).

Reply:
Okay, an idea: what comes after the comment "#version of the function semistandard_tableaux(shape::Vector{T}, max_val=sum(shape))" in `schur_polynomial` is actually an interator over semistandard tableaux. This should (later) go into an actual separate iterator. Then this function looks cleaner. Moreover, one can reuse it do implement `schur_polynomial` with argument a vector x of ring elements which returns the Schur polynomial evaluated at this vector but which should be faster than first building a large polynomial and then evaluation. One application I have in mind is the q-analog where one evaluates at 1,q,q^2,... (but there's a closed formula for the result actually).